minard/RSTools.py

```diff
@@ -0,0 +1,484 @@
+    # are downloaded, or there are no more tables to download, or reach a hard cap in number of loops (just in case).
+    num_loops = 0
+    while (len(run_numbers) <= (offset + limit)) and (no_more_tables == False) and (num_loops <= 100):
+        earliest_run = run_numbers[-1]
+        new_filtered_rs_tables, new_run_numbers, no_more_tables = get_filtered_RS_tables(earliest_run-1, offset, limit, result, criteria)
+        filtered_rs_tables.update(new_filtered_rs_tables)
+        run_numbers += new_run_numbers
+        num_loops += 1
+
+    # Only keep the runs that will be listed on the page
+    final_rs_tables = OrderedDict()
+    for i in range(offset, (offset+limit)):
+        final_rs_tables[run_numbers[i]] = filtered_rs_tables[run_numbers[i]]
+
+    return final_rs_tables
+
+############ RUNSELECTION_RUN PAGE FUNCTIONS ############
+
+def get_criteria_tables(runNum, crit_version):
+    '''Get criteria tables associated with run, list of criteria tags.
+    crit_version is a dictionary with a key for every criteria tag, and
+    the associate value is the preferred version number'''
+
+    # Download criteria tables that match run number and criteria tag(s)
+    query = "SELECT meta_data, timestamp FROM run_selection"
+    query += " WHERE run_min <= {} AND (run_max IS NULL OR run_max >= {}) AND type = 'CRITERIA'".format(int(runNum), int(runNum))
+    query += " AND ("
+    for criteria in crit_version:
+        query += "criteria = '{}' OR ".format(criteria)
+    query = query[:-4]  # Remove last ' OR '
+    query += ") ORDER BY timestamp DESC LIMIT 50"
+
+    conn = engine_nl.connect()
+    resultQuery = conn.execute(query)
+
+    table_list = []
+    for row in resultQuery.fetchall():
+        tempt_dict = {}
+        tempt_dict['meta_data'] = row[0]
+        tempt_dict['timestamp'] = row[1]
+        table_list.append(tempt_dict)
+
+    # Repackage and check for duplicates
+    crit_tables = {}
+    for table in table_list:
+        criteria = table['meta_data']['index']
+        if criteria not in crit_tables and criteria in crit_version:
+            crit_tables[criteria] = table
+        else:
+            # There is a duplicate
+            if decide_replace_table(crit_tables[criteria], table, crit_version[criteria]):
+                crit_tables[criteria] = table
+
+    # Check if we got a criteria table for each inputted criteria
+    for crit in crit_version:
+        if crit not in crit_tables:
+            print('WARNING: No criteria table found for {} criteria.'.format(crit))
+            crit_tables[crit] = False
+
+    return crit_tables
+
+def format_general_info(rs_tables, criteria_list):
+    general_info = OrderedDict()
+    first_table = rs_tables[criteria_list[0]]  # Take first table for general information (no particular reason)
+
+    general_info['Run'] = first_table['meta_data']['run_range'][0]
+    general_info['Start timestamp'] = first_table['meta_data']['run_time']['notes']['dt']['timestamp']
```

Comment:
Can you format 'Start timestamp' and 'Last reviewed' to remove the nanoseconds? It is a little hard to read on the webpage in the current format.